extensions/authentication-passport/src/__tests__/acceptance/authentication-with-passport-strategy-oauth2-adapter.acceptance.ts

```diff
@@ -0,0 +1,194 @@
+import {get, param} from '@loopback/openapi-v3';
+import {
+  Client,
+  createClientForHandler,
+  expect,
+  supertest
+} from '@loopback/testlab';
+import {RestApplication, RedirectRoute} from '@loopback/rest';
+import {startApp as startMockProvider, stopApp as stopMockProvider} from './fixtures/oauth2-provider';
+import * as url from 'url';
+import { inject } from '@loopback/core';
+
+/**
+ * options to pass to the Passport Strategy
+ */
+const oauth2Options: StrategyOptions = {
+  clientID: '1111',
+  clientSecret: '1917e2b73a87fd0c9a92afab64f6c8d4',
+  callbackURL: 'http://localhost:8080/auth/thirdparty/callback',
+  authorizationURL: 'http://localhost:9000/oauth/dialog',
+  tokenURL: 'http://localhost:9000/oauth/token',
+}
+
+/**
+ * verify function for the oauth2 strategy
+ * This function mocks a lookup against a user profile datastore
+ * 
+ * @param accessToken 
+ * @param refreshToken 
+ * @param profile 
+ * @param done 
+ */
+const verify: VerifyFunction = function (accessToken: string, refreshToken: string, profile: any, done: VerifyCallback) {
+  const userProfile: MyUser = profile as MyUser;
+  let user: UserProfile = userRepository.findUser(userProfile.id);
+  console.log('verify', profile, user);
+  if (!user) {
+    return done(new Error('user not found'));
+  }
+  user.token = accessToken;
+  return done(null, user);
+}
+
+/**
+ * convert user info to user profile
+ * @param user 
+ */
+const myUserProfileFactory: UserProfileFactory<MyUser> = function(
+  user: MyUser,
+): UserProfile {
+  const userProfile = {[securityId]: user.id};
+  return userProfile;
+};
+
+/**
+ * Login controller for third party oauth provider
+ * 
+ * This creates an authentication endpoint for the third party oauth provider
+ * 
+ * Two methods are expected
+ * 
+ * 1. loginToThirdParty 
+ *           i. an endpoint for api clients to login via a third party app
+ *          ii. the passport strategy identifies this call as a redirection to third party
+ *         iii. this endpoint redirects to the third party authorization url
+ * 
+ * 2. thirdPartyCallBack
+ *           i. this is the callback for the thirdparty app
+ *          ii. on successful user login the third party calls this endpoint with an access code
+ *         iii. the passport oauth2 strategy exchanges the code for an access token
+ *          iv. the passport oauth2 strategy then calls the provided `verify()` function with the access token
+ */
+export class Oauth2Controller {
+  constructor() {}
+
+  // this configures the oauth2 strategy
+  @authenticate('oauth2')
+  // we have modeled this as a GET endpoint
+  @get('/auth/thirdparty')
+  // loginToThirdParty() is the handler for '/auth/thirdparty'
+  // this method is injected with 'x-loopback-authentication-redirect-url'
+  // the value for 'x-loopback-authentication-redirect-url' is set by the passport strategy adapter
+  loginToThirdParty(@param.query.string('x-loopback-authentication-redirect-url') redirectUrl: string,
```

Comment:
I wonder if we can inject such information from `RequestContext` instead of query string.